### PR TITLE
refactor: rename packages and files in dagger-common serde

### PR DESCRIPTION
**Pull request by Meghajit**
Solves for https://github.com/odpf/dagger/issues/139

### CODE REVIEW

**Comment by gauravsinghania** [comment]
Can we please remove `type` from different handlers and let be there only for `TypeHandler`. At present we have 2 interfaces `TypeHandler` and `PrimitiveHandler`. However, when classes extend from both of them they are referred as `MapTypeHandler` and `StringTypeHandler` which is little confusing. Rather if we just have above interfaces and classes are `MapHandler` and `StringHandler` it would be self implied.

**Comment by Meghajit** [comment]
> Can we please remove `type` from different handlers and let be there only for `TypeHandler`. At present we have 2 interfaces `TypeHandler` and `PrimitiveHandler`. However, when classes extend from both of them they are referred as `MapTypeHandler` and `StringTypeHandler` which is little confusing. Rather if we just have above interfaces and classes are `MapHandler` and `StringHandler` it would be self implied.

Fixed as part of https://github.com/odpf/dagger/pull/138/commits/369b7bff53e131cc10a501d9469ec83b1185b082

File: dagger-common/src/main/java/io/odpf/dagger/common/serde/typehandler/TypeHandlerFactory.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class TypeHandlerFactory {
+    private static Map<String, TypeHandler> protoHandlerMap = new ConcurrentHashMap<>();
+
+    /**
+     * Gets proto handler.
+     *
+     * @param fieldDescriptor the field descriptor
+     * @return the proto handler
+     */
+    public static TypeHandler getProtoHandler(final Descriptors.FieldDescriptor fieldDescriptor) {
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply dagger-common/src/main/java/io/odpf/dagger/common/serde/typehandler/TypeHandlerFactory.java:31]: Let's rename it to `getTypeHandler`.
@prakharmathur82 Will become data type agnostic then. WDYT ?

File: dagger-common/src/main/java/io/odpf/dagger/common/serde/typehandler/TypeHandlerFactory.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import io.odpf.dagger.common.serde.typehandler.repeated.RepeatedPrimitiveHandler;
+import io.odpf.dagger.common.serde.typehandler.repeated.RepeatedStructMessageHandler;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The factory class for Proto handler.
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply dagger-common/src/main/java/io/odpf/dagger/common/serde/typehandler/TypeHandlerFactory.java:20]: It's java doc from the old code. Will fix.

File: dagger-common/src/main/java/io/odpf/dagger/common/serde/typehandler/TypeHandlerFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (prakharmathur82) [reply dagger-common/src/main/java/io/odpf/dagger/common/serde/typehandler/TypeHandlerFactory.java:31]: sure

[the remark this answers is not in the document]
Comment (Meghajit) [reply dagger-common/src/main/java/io/odpf/dagger/common/serde/typehandler/TypeHandlerFactory.java:31]: Will fix shortly

[the remark this answers is not in the document]
Comment (Meghajit) [reply dagger-common/src/main/java/io/odpf/dagger/common/serde/typehandler/TypeHandlerFactory.java]: @prakharmathur82 Fixed as part of https://github.com/odpf/dagger/pull/138/commits/30146a5af63600308b86eb2815ccc1e0aa443c11

**Pull request merged**